crafting-dead-immerse/src/main/java/com/craftingdead/immerse/world/level/block/BaseCenterBlock.java

```diff
@@ -58,4 +63,16 @@ public void onRemove(BlockState blockState, Level level, BlockPos blockPos,
     }
     super.onRemove(blockState, level, blockPos, newState, moved);
   }
+
+  @Override
+  public boolean onDestroyedByPlayer(BlockState state, Level level, BlockPos pos, Player player,
+      boolean willHarvest, FluidState fluid) {
+    if (!level.isClientSide()) {
```

Comment:
Could change this into `player instanceof ServerPlayer player` to avoid the cast to `ServerPlayer` later on.